### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mira1zu**
- [DEMO LINK](https://mira1zu.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by AndriiNychaiuk — APPROVED** [review]
👍

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +17,160 @@ const goodsFromServer: string[] = [
...
+class App extends React.PureComponent<{}, State> {
+  state: State = {
+    initialGoods: goodsFromServer,
+    isStarted: true,
+    isReversed: false,
+    sortBy: 'reset',
+    minLength: 1,
+  };
+
+  onStartButtonClick = () => {
```
Comment (Olexandr6) [inline src/App.tsx:37]: This naming is suitable for a component props, not for handler functions

Read more: https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +17,160 @@ const goodsFromServer: string[] = [
...
+          return firstProduct.length - secondProduct.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      currentGoods.reverse();
+    }
```
Comment (Olexandr6) [inline src/App.tsx:98]: This logic shouldn't be here. Method `render` should be as clear as possible.
Let's move this logic to a separate method

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +17,160 @@ const goodsFromServer: string[] = [
...
+                    <select
+                      name="minLength"
+                      onChange={({ target }) => {
+                        this.setState({
+                          minLength: +target.value,
+                        });
+                      }}
+                    >
+                      {[...new Array(10)].map((_, i) => (
+                        // eslint-disable-next-line react/no-array-index-key
```
Comment (Olexandr6) [inline src/App.tsx:161]: 🥲

File: src/App.tsx [hunk 4]
```diff
@@ -14,11 +17,160 @@ const goodsFromServer: string[] = [
...
+                    onClick={this.onSortByLengthButtonClick}
+                  >
+                    Sort by length
+                  </button>
+                  <div className="select is-primary is-fullwidth">
+                    <select
+                      name="minLength"
+                      onChange={({ target }) => {
+                        this.setState({
+                          minLength: +target.value,
```
Comment (Olexandr6) [inline src/App.tsx:156]:
```suggestion
                          minLength: Number(target.value),
```

**Review by Olexandr6 — CHANGES_REQUESTED** [review]
Good solution, let's work on impovements

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**